pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -124,4 +136,215 @@ var _ = Describe("Masquerade infrastructure configurator", func() {
+			podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: mtu}}
+		})
+
+		BeforeEach(func() {
+			ipv6GwAddr, _ := netlink.ParseAddr(ipv6GwStr)
+			ipv6VmAddr, _ := netlink.ParseAddr(vmIPv6Str)
+			dhcpConfig = &cache.DHCPConfig{
+				Name:                ifaceName,
+				IP:                  podIP,
+				IPv6:                *ipv6VmAddr,
+				Mtu:                 uint16(mtu),
+				AdvertisingIPAddr:   gatewayAddr.IP.To4(),
+				AdvertisingIPv6Addr: ipv6GwAddr.IP,
+			}
+
+			masqueradeConfigurator.podNicLink = podLink
+			masqueradeConfigurator.vmGatewayAddr = gatewayAddr
+			masqueradeConfigurator.vmIPv4Addr = podIP
+			masqueradeConfigurator.vmGatewayIpv6Addr = ipv6GwAddr
+			masqueradeConfigurator.vmIPv6Addr = *ipv6VmAddr
+		})
+
+		When("the pod features a properly configured primary link", func() {
+			BeforeEach(func() {
+				handler.EXPECT().LinkAdd(inPodBridge).Return(nil)
+				handler.EXPECT().LinkSetUp(inPodBridge).Return(nil)
+				handler.EXPECT().DisableTXOffloadChecksum(inPodBridge.Name).Return(nil)
+				handler.EXPECT().CreateTapDevice(tapDeviceName, queueCount, launcherPID, mtu, netdriver.LibvirtUserAndGroupId).Return(nil)
+				handler.EXPECT().BindTapDeviceToBridge(tapDeviceName, inPodBridge.Name).Return(nil)
+			})
+
+			table.DescribeTable("should work with", func(backend netfilterBackend, additionalIPProtocol ...iptables.Protocol) {
+				mockVML3Config(*masqueradeConfigurator, ifaceName, inPodBridge, additionalIPProtocol...)
+				mockNATNetfilterRules(handler, *dhcpConfig, backend, additionalIPProtocol...)
+
+				Expect(masqueradeConfigurator.PreparePodNetworkInterface()).To(Succeed())
+			},
+				table.Entry("NFTables backend on an IPv4 cluster", nft),
+				table.Entry("IPTables backend on an IPv4 cluster", ipTables),
+				table.Entry("NFTables backend on a dual stack cluster", nft, iptables.ProtocolIPv6),
+				table.Entry("IPTables backend on a dual stack cluster", ipTables, iptables.ProtocolIPv6),
+			)
+		})
+	})
 })
+
+func mockVML3Config(configurator MasqueradePodNetworkConfigurator, podIface string, inPodBridge *netlink.Bridge, optionalIPProtocol ...iptables.Protocol) {
+	protocols := getProtocols(optionalIPProtocol...)
+	hasIPv6Config := len(protocols) > 1
+	mockedHandler := configurator.handler.(*netdriver.MockNetworkHandler)
+	mockedHandler.EXPECT().IsIpv6Enabled(podIface).Return(hasIPv6Config, nil).Times(2) // once on create bridge, another on prepare pod network
+
+	for _, l3Protocol := range protocols {
+		gatewayAddr := configurator.vmGatewayAddr
+		if l3Protocol == iptables.ProtocolIPv6 {
+			gatewayAddr = configurator.vmGatewayIpv6Addr
+		}
+		mockedHandler.EXPECT().AddrAdd(inPodBridge, gatewayAddr).Return(nil)
+	}
+}
+
+func mockNATNetfilterRules(handler *netdriver.MockNetworkHandler, dhcpConfig cache.DHCPConfig, netfilterBackend netfilterBackend, optionalIPProtocol ...iptables.Protocol) {
+	getNFTIPString := func(proto iptables.Protocol) string {
```

Comment:
Please make it a regular function.

Reply:
So we will not need to investigate if it captures an external variable or not.
I will not insist on this.

Reply:
Cool.